callbacks/preload.go

```diff
@@ -125,13 +125,15 @@ func preloadEntryPoint(db *gorm.DB, joins []string, relationships *schema.Relati
 				case reflect.Slice, reflect.Array:
 					if rv.Len() > 0 {
 						reflectValue := rel.FieldSchema.MakeSlice().Elem()
-						reflectValue.SetLen(rv.Len())
 						for i := 0; i < rv.Len(); i++ {
 							frv := rel.Field.ReflectValueOf(db.Statement.Context, rv.Index(i))
 							if frv.Kind() != reflect.Ptr {
-								reflectValue.Index(i).Set(frv.Addr())
+								reflectValue = reflect.Append(reflectValue, frv.Addr())
 							} else {
-								reflectValue.Index(i).Set(frv)
+								if frv.IsNil() {
+									continue
```

Comment:
Do we need special handling if `reflectValue.Len` is 0?

Reply:
I guess not, I added another unit test `TestJoinsPreload_Issue7013_NoEntries` and checked with a conditional breakpoint that we reach the case where `rv.Len` is 0, and we correctly get an empty list of objects (without panic of course)